lib/sse_parser.ex

```diff
@@ -37,7 +37,7 @@ defmodule MarathonEventExporter.SSEParser do
   Feeds some input data into the parser.
   """
   def feed_data(server, data) do
-    GenServer.cast(server, {:feed_data, data})
+    GenServer.call(server, {:feed_data, data})
```

Comment:
It's a GenServer because it has its own state and I wanted to keep it separate from the more generic HTTP stream handling.

I mostly changed `feed_data` from `cast` to `call` so that we don't have to hack around the asynchrony in the tests.

Reply:
I guess I don't know when `GenServer` is appropriate, or even what the other options are 😅. But seems fine for now.